docs/modules/ROOT/pages/utilities.adoc

```diff
@@ -0,0 +1,43 @@
+= Utilities
+
+The OpenZeppelin Stylus Contracts provide a ton of useful utilities that you can use in your project.
+For a complete list, check out the https://docs.rs/openzeppelin_stylus/utils/index.html[API Reference].
+Here are some of the more popular ones.
+
+[[introspection]]
+== Introspection
+
+As well as in Solidity, at Stylus, it's frequently helpful to know whether a contract supports an interface you'd like to use.
+https://eips.ethereum.org/EIPS/eip-165[`ERC-165`] is a standard that helps do runtime interface detection.
+Contracts provide helpers both for implementing ERC-165 in your contracts and querying other contracts:
+
+* https://docs.rs/openzeppelin_stylus/utils/introspection/erc165/trait.IErc165.html[`IERC165`] — this is the ERC-165 trait that defines https://docs.rs/openzeppelin_stylus/utils/introspection/erc165/trait.IErc165.html#tymethod.supports_interface[`supportsInterface`]. In order to implement ERC-165 interface detection, you should manually expose https://docs.rs/openzeppelin_stylus/utils/introspection/erc165/trait.IErc165.html#tymethod.supports_interface[`supportsInterface`] function in your contract.
+
+[source,rust]
+----
+sol_storage! {
+    #[entrypoint]
+    struct Erc721Example {
+        #[borrow]
+        Erc721 erc721;
+    }
+}
+
+#[public]
+#[inherit(Erc721)]
+impl Erc721Example {
+    pub fn supports_interface(interface_id: FixedBytes<4>) -> bool {
+        Erc721::supports_interface(interface_id)
+    }
+}
+
+----
+
+[[structures]]
+== Structures
+
+Some use cases require more powerful data structures than arrays and mappings offered natively in Stylus sdk.
+Contracts provides these libraries for enhanced data structure management:
```

Comment:
I think we should name it Contracts for Stylus to be more consistent with the name of the library. Alternatively I'm happy to use this crate, or this library....
```suggestion
Contracts for Stylus provides these libraries for enhanced data structure management:
```